### PR TITLE
aber is not a filler when used as a conjunction

### LINKED ISSUE #484: ", aber" false positives
"Aber" is a filler only in cases when it is not used as a conjunction.

At a formal level, we can identify conjunctions by the surrounding punctuation.

Na gut, wir gehen einkaufen, aber wir kaufen keine Süßigkeiten.

Aber nachdem es so lange geregnet hatte, war der Weg zu matschig, um spazieren zu gehen.

### PR DESCRIPTION
**Pull request by lsmith77**
fixes https://github.com/witty-works/nlp_api/issues/484

also removed some duplicate code

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1423,6 +1423,13 @@ def gendered_denom_analysis_de(
                 if c_doc[i].lemma_ == word:
```
Comment (lsmith77) [inline app/main.py]: @lnazarenko we seem to be missing a test case for this code path, can you help me construct one?

File: app/main.py [hunk 2]
```diff
@@ -1423,6 +1423,13 @@ def gendered_denom_analysis_de(
...
                     c_doc_morph_number = c_doc[i].morph.get("Number")
                     if is_number_list_empty(c_doc_morph_number, c_doc[i], full_text):
+                        alternative = alternative_all
```
Comment (lsmith77) [inline app/main.py]: @lnazarenko we seem to be missing a test case for this code path, can you help me construct one?

File: app/main.py [hunk 3]
```diff
@@ -1585,45 +1555,30 @@ def style_word_analysis_de(
...
-                                tokens[i].text,
-                                full_text,
-                                category,
-                                subcategory,
-                                tokens[i].idx,
-                                tokens[i].idx + len(tokens[i].text),
-                                alternative,
-                            )
-                        )
+                continue
```
Comment (lsmith77) [inline app/main.py:1527]: @lnazarenko we seem to be missing a test case for this code path, can you help me construct one?

**Comment by lsmith77** [comment]
@lnazarenko this is now a rather large refactoring of the code, so we should probably review this together when we are sitting next to each other. when are you in the office next?

File: app/main.py [hunk 4]
```diff
@@ -1298,68 +1302,38 @@ def agentic_language_analysis_de(
     category,
 ):
     list_tokens = []
-    dic_anc = {}
-    list_false_positives = []
     for token in tokens:
-        # check if the user query have false positives
```
Comment (lsmith77) [inline app/main.py:1304]: removed agentic false positives

File: app/main.py [hunk 5]
```diff
@@ -1382,56 +1356,25 @@ def ub_words_phrase_matcher_de(
     patterns = [model[lang.lang].make_doc(text) for text in list(df_sentence["Lemma"])]
     matcher.add("TerminologyList", patterns)
 
-    dic_anc = {}
-    list_false_positives = []
```
Comment (lsmith77) [inline app/main.py:1386]: removed agentic false positives

File: app/main.py [hunk 6]
```diff
@@ -1484,150 +1433,69 @@ def gendered_denom_analysis_de(
...
-                                    c_doc[i].text,
-                                    full_text,
-                                    category,
-                                    subcategory,
-                                    c_doc[i].idx,
-                                    None,
-                                    alternative_plur,
-                                )
-                            )
+        tokens = Doc.from_docs(docs)
```
Comment (lsmith77) [inline app/main.py:1436]: removed duplicate code

**Review by lnazarenko — APPROVED** [review]

**Pull request merged**